core/src/main/java/io/smallrye/openapi/api/models/media/SchemaImpl.java

```diff
@@ -891,7 +1078,421 @@ public Schema addOneOf(Schema oneOf) {
     @Override
     public void removeOneOf(Schema oneOf) {
         incrementModCount();
-        ModelUtil.remove(this.oneOf, oneOf);
+        removeFromListProperty(PROP_ONE_OF, oneOf);
+    }
+
+    @Override
+    public Map<String, Object> getExtensions() {
+        Map<String, Object> extensions = new HashMap<>();
+        data.forEach((k, v) -> {
+            if (k.startsWith("x-")) {
+                extensions.put(k, v);
+            }
+        });
+        return extensions.isEmpty() && !explicitSetExtensions ? null : extensions;
+    }
+
+    @Override
+    public Schema addExtension(String name, Object value) {
+        setProperty(name, value);
+        explicitSetExtensions = true;
+        return this;
+    }
```

Comment:
The new mutator methods are missing calls to `incrementModCount`. I wonder if it would be better to move `modCount` up to `MapBasedModelImpl` as protected and consolidate usage of `incrementModCount`. Either way is fine with me.

Reply:
Good spot, I've pushed up changes moving the `incrementModCount()` calls into the `setProperty` methods on `SchemaImpl`.

I've kept it all in `SchemaImpl` for the moment since we already have additional logic we do before getting and setting properties on a schema.